shop/src/main/java/ro/msg/learning/shop/strategy/SingleLocationOrder.java

```diff
@@ -0,0 +1,89 @@
+package ro.msg.learning.shop.strategy;
+
+import lombok.AllArgsConstructor;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+import ro.msg.learning.shop.domain.Order;
+import ro.msg.learning.shop.domain.OrderDetail;
+import ro.msg.learning.shop.domain.Stock;
+import ro.msg.learning.shop.domain.key.OrderDetailId;
+import ro.msg.learning.shop.domain.key.StockId;
+import ro.msg.learning.shop.dto.CreateOrderDto;
+import ro.msg.learning.shop.dto.OrderDto;
+import ro.msg.learning.shop.exception.BadRequestException;
+import ro.msg.learning.shop.mapper.OrderMapper;
+import ro.msg.learning.shop.repository.OrderRepository;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.repository.StockRepository;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static ro.msg.learning.shop.message.Messages.*;
+
+
+@AllArgsConstructor
+@Component
+@ConditionalOnProperty(name = "${strategy}", havingValue = "abundant")
+public class SingleLocationOrder implements OrderLocationStrategy {
+
+    private final StockRepository stockRepository;
+    private final ProductRepository productRepository;
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+
+
+    @Override
+    public OrderDto create(CreateOrderDto createOrderDto) throws BadRequestException {
+        List<UUID> firstProductLocations = stockRepository.findSuitableLocation(createOrderDto.getOrderProductDtoList().get(0).getProductId(), createOrderDto.getOrderProductDtoList().get(0).getQuantity());
+
+        for (int i = 0; i < createOrderDto.getOrderProductDtoList().size(); i++) {
```

Comment:
We should use nicer for statements in general like:
` for (OrderProductDto orderProductDto : orderProductDtoList){...}`

Reply:
You can extract the list of products at the beginning because you will use it in many places.